CHANGES.md

```diff
@@ -1,25 +1,23 @@
 # 2.14.0 - 29 May 2024
 
-This release fixes the `RelationDataContent.update` method to follow `dict.update` semantics, that is allow both updating with another dict, an iterable, keyword arguments or a mixture thereof.
```

Comment:
I think all of these changes are fair enough, but this one we did deliberately: we thought it was a good precedent to include the summary paragraph in CHANGES.md moving forward (though no need to add them to the others retroactively). What do you think?

Reply:
As a meta note: I think we should definitely update the instructions in HACKING.md if we're going to make changes like that, so that we can be consistent and all understand what the expectations are (particularly when it's counter to something that's already been discussed as a team). Changing decisions is fine, but best done more openly.

I think it depends on what we see as the purpose of this file, particularly compared to the GtiHub releases page. This tangentially relates to Dora's suggestion to link to the releases page from ops.readthedocs.io, too.

If CHANGES.md primarily exists so that the changes are present in artefacts (both the GitHub tarballs and the PyPI sdist) then having it closer to the GitHub releases (including the intro section) makes sense. If we do this then I think we probably also want to change the project.urls link to point to the GitHub releases page rather than to CHANGES.md.

If it exists to (also) be more of a reference - e.g. something to scan through to check for anything that you need to be aware of when upgrading, then I think a plainer log of changes makes sense. In this form, it's a mid point between `git log` and the releases. This is how I used changelogs in the past, but these days I'm most often seeing dependency bumps with Dependabot and that pulls in the release notes (at least for packages with GitHub releases, like ops).

Looking around at popular Python packages, they are almost all bulleted lists (logs) of changes, without any sort of preamble that you find in release announcements. My inclination is to fit with the ecosystem's pattern.

Reply:
Good point about updating `HACKING.md` if we do this.

> Looking around at popular Python packages, they are almost all bulleted lists (logs) of changes, without any sort of preamble that you find in release announcements. My inclination is to fit with the ecosystem's pattern.

Fair enough. Let's run with that.